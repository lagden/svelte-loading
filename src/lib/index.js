export {default as Loading} from './Loading.svelte'
export * as acts from './acts.js'
