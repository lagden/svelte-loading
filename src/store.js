import {writable} from 'svelte/store'

export const isloading = writable(false)
