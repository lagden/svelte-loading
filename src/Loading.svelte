<script>
	import {fade} from 'svelte/transition'
	import {isloading} from './store'
</script>

{#if $isloading}
	<div class="_tadashi_svelte_loading" transition:fade|local={{duration: 400}}>
		<div class="_tadashi_svelte_loading_animation"></div>
	</div>
{/if}

<style type="text/css">
	:root {
		--tadashi_svelte_loading_background_color: hsla(0, 0%, 0%, 0.3);
		--tadashi_svelte_loading_background_image: none;

		--tadashi_svelte_loading_animation_padding: 15px;
		--tadashi_svelte_loading_animation_border: 6px solid hsl(0, 0%, 80%);
		--tadashi_svelte_loading_animation_border_right_color: hsl(0, 0%, 50%);
		--tadashi_svelte_loading_animation_border_radius: 22px;
		--tadashi_svelte_loading_animation_duration: 0.5s;
	}

	._tadashi_svelte_loading {
		background-color: var(--tadashi_svelte_loading_background_color);
		background-image: var(--tadashi_svelte_loading_background_image);
		position: fixed;
		top: 0;
		left: 0;
		width: 100vw;
		height: 100vh;
	}

	._tadashi_svelte_loading_animation {
		height: 0;
		width: 0;
		padding: var(--tadashi_svelte_loading_animation_padding);
		border: var(--tadashi_svelte_loading_animation_border);
		border-right-color: var(--tadashi_svelte_loading_animation_border_right_color);
		border-radius: var(--tadashi_svelte_loading_animation_border_radius);
		animation: _tadashi_svelte_loading_rotate var(--tadashi_svelte_loading_animation_duration) infinite linear;
		position: absolute;
		left: 50%;
		top: 50%;
	}

	@keyframes _tadashi_svelte_loading_rotate {
		100% {
			transform: rotate(360deg);
		}
	}
</style>
