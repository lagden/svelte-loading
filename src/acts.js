import {writable} from 'svelte/store'

const isloading = writable(false)
const show = value => {
	isloading.set(value)
}


export {
	isloading,
	show,
}
