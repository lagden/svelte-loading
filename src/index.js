/* istanbul ignore file */

import * as acts from './acts'
import Loading from './Loading.svelte'

export {
	Loading,
	acts
}
