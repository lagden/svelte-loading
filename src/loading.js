/* istanbul ignore file */

import * as acts from './acts.js'
import Loading from './Loading.svelte'

export {
	Loading,
	acts,
}
