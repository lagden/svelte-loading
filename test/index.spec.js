/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Loading, acts} from '../src'

describe('Loading', () => {
	afterEach(cleanup)

	test('should match snapshot', () => {
		acts.show(true)

		const {container} = render(Loading, {
			props: {
				class: 'teste',
				'data-theme': 'dark'
			}
		})
		expect(container).toMatchSnapshot()
	})
})
