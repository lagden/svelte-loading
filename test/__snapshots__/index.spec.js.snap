// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match snapshot 1`] = `
<body>
  <div>
    <div
      class="_tadashi_svelte_loading svelte-s3ye64"
    >
      <div
        class="_tadashi_svelte_loading_animation svelte-s3ye64"
      />
    </div>
    
  </div>
</body>
`;
